### PR TITLE
feat: Add support for multi arch images build

### PR DESCRIPTION
**Pull request by e0ne**
Add support for multi arch images build

### CODE REVIEW

**Review by ykulazhenkov — APPROVED** [review]

**Review by rollandf — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -15,7 +15,8 @@
 # Build the manager binary
 FROM golang:1.20 as builder
 
-ARG TARGETARCH=amd64
+ARG ARCH ?= $(shell go env GOARCH)
```
Comment (killianmuldoon) [inline Dockerfile:18]: If I understand right multi-arch builds are then done somewhere else - this change is just enabling them. 

Could we make the docker build in the Makefile multi-arch? Or is it better to just have that as a detail of the release process?

**Review by almaslennikov — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (e0ne) [reply Dockerfile:18]: @killianmuldoon I'll add a note into README to to build such images. I don't want to change make target to not complicate develpment process

**Review by rollandf — CHANGES_REQUESTED** [review]
`make image` does not work now with this Dockerfile, seems to break some CI.

**Comment by rollandf** [comment]
> `make image` does not work now with this Dockerfile, seems to break some CI.

Seems podman related issue

**Review by rollandf — APPROVED** [review]

**Review by rollandf — APPROVED** [review]

**Pull request merged**